core/services/vrf/v2/integration_v2_plus_test.go

```diff
@@ -0,0 +1,1560 @@
+	t.Parallel()
+	testEoa(t, false)
+}
+
+func TestVRFV2PlusIntegration_SingleConsumer_EOA_Request_Batching_Enabled(t *testing.T) {
+	t.Parallel()
+	testEoa(t, true)
+}
+
+func TestVRFV2PlusIntegration_SingleConsumer_EIP150_HappyPath(t *testing.T) {
+	t.Parallel()
+	callBackGasLimit := int64(2_500_000)            // base callback gas.
+	eip150Fee := callBackGasLimit / 64              // premium needed for callWithExactGas
+	coordinatorFulfillmentOverhead := int64(90_000) // fixed gas used in coordinator fulfillment
+	gasLimit := callBackGasLimit + eip150Fee + coordinatorFulfillmentOverhead
+
+	key1 := cltest.MustGenerateRandomKey(t)
+	gasLanePriceWei := assets.GWei(10)
+	config, _ := heavyweight.FullTestDBV2(t, "vrfv2_singleconsumer_eip150_happypath", func(c *chainlink.Config, s *chainlink.Secrets) {
+		simulatedOverrides(t, assets.GWei(10), v2.KeySpecific{
+			// Gas lane.
+			Key:          ptr(key1.EIP55Address),
+			GasEstimator: v2.KeySpecificGasEstimator{PriceMax: gasLanePriceWei},
+		})(c, s)
+		c.EVM[0].GasEstimator.LimitDefault = ptr(uint32(gasLimit))
+		c.EVM[0].MinIncomingConfirmations = ptr[uint32](2)
+	})
+	ownerKey := cltest.MustGenerateRandomKey(t)
+	uni := newVRFCoordinatorV2PlusUniverse(t, ownerKey, 1)
+	app := cltest.NewApplicationWithConfigV2AndKeyOnSimulatedBlockchain(t, config, uni.backend, ownerKey, key1)
+	consumer := uni.vrfConsumers[0]
+	consumerContract := uni.consumerContracts[0]
+	consumerContractAddress := uni.consumerContractAddresses[0]
+	// Create a subscription and fund with 500 LINK.
+	subAmount := big.NewInt(1).Mul(big.NewInt(5e18), big.NewInt(100))
+	subID := subscribeAndAssertSubscriptionCreatedEvent(t, consumerContract, consumer, consumerContractAddress, subAmount, uni.rootContract, uni.backend)
+
+	// Fund gas lane.
+	sendEth(t, ownerKey, uni.backend, key1.Address, 10)
+	require.NoError(t, app.Start(testutils.Context(t)))
+
+	// Create VRF job.
+	jbs := createVRFJobs(
+		t,
+		[][]ethkey.KeyV2{{key1}},
+		app,
+		uni.rootContract,
+		uni.rootContractAddress,
+		uni.batchCoordinatorContractAddress,
+		uni.coordinatorV2UniverseCommon,
+		nil,
+		vrfcommon.V2Plus,
+		false,
+		gasLanePriceWei)
+	keyHash := jbs[0].VRFSpec.PublicKey.MustHash()
+
+	// Make the first randomness request.
+	numWords := uint32(1)
+	requestRandomnessAndAssertRandomWordsRequestedEvent(t, consumerContract, consumer, keyHash, subID, numWords, uint32(callBackGasLimit), uni.rootContract, uni.backend)
+
+	// Wait for simulation to pass.
+	gomega.NewGomegaWithT(t).Eventually(func() bool {
+		uni.backend.Commit()
+		runs, err := app.PipelineORM().GetAllRuns()
+		require.NoError(t, err)
+		t.Log("runs", len(runs))
+		return len(runs) == 1
+	}, testutils.WaitTimeout(t), time.Second).Should(gomega.BeTrue())
+
+	t.Log("Done!")
+}
+
+func TestVRFV2PlusIntegration_SingleConsumer_EIP150_Revert(t *testing.T) {
+	t.Parallel()
+	callBackGasLimit := int64(2_500_000)            // base callback gas.
```

Comment:
I wonder if this test can be pulled out into a function (like the previous tests) and called with the V2/V2Plus version from the V2 and the V2Plus test files respectively?

Reply:
yep refactored several tests like this 👍